### PR TITLE
Community sidechain branch update - teerex followup, pallet-encointer-reputations-commitment

### PR DESCRIPTION
**Pull request by OverOrion**
Merge the previous one  https://github.com/encointer/encointer-node/pull/334 first. (Will do a rebase after that).

Companion of https://github.com/integritee-network/integritee-node/pull/180.

This PR adds the following two required things into the `community-sidechain` branch (which are needed for the Personhood Oracle)
-  pallet-encointer-reputation-commitments (PO depends on this)
- teerex pallet update (because there were some breaking changes)

### CODE REVIEW

File: client/Cargo.toml [hunk 1]
```diff
@@ -26,6 +26,7 @@ encointer-primitives = { git = "https://github.com/encointer/pallets", branch =
 pallet-encointer-balances = { git = "https://github.com/encointer/pallets", branch = "polkadot-v0.9.42" }
 pallet-encointer-ceremonies = { git = "https://github.com/encointer/pallets", branch = "polkadot-v0.9.42" }
 pallet-encointer-communities = { git = "https://github.com/encointer/pallets", branch = "polkadot-v0.9.42" }
+pallet-encointer-reputation-commitments = { default-features = false, git = "https://github.com/encointer/pallets", branch = "polkadot-v0.9.42" }
```
Comment (clangenb) [inline client/Cargo.toml]: Should not have default-features false.

**Review by clangenb — COMMENTED** [review]
LGTM, I guess the one thing I noted, is an issue on master branch too anyhow.

File: client/Cargo.toml (hunk 1 above)
Comment (OverOrion) [reply client/Cargo.toml]: It seems that is not even on master, I added it there by mistake, nice catch!

**Review by brenzi — APPROVED** [review]
LGTM

**Comment by OverOrion** [comment]
[Currently the CI fails at the `bootstrap_demo_community` step:
](https://github.com/encointer/encointer-node/actions/runs/5818366183/job/15776167346#step:12:124)

```
    if(not client.balance(faucet_account) == balance(10000)):
        print(f"Wrong Faucet balance after faucet creation")
        exit(1)
```

I'll add some logs to find why it suddenly fails. In the meantime can one of you @clangenb @brenzi have a look at it and maybe you'll have some idea where it stems from?

**Comment by OverOrion** [comment]
~~Seems like the CI checks out the `master` branch for the `integration-test` part. I'll have to overwrite it for now.~~
EDIT: The older CI results were open.

**Comment by brenzi** [comment]
I wouldn't expect this error based on your changes. I can look into it

**Comment by brenzi** [comment]
the faucet test fails because meanwhile the create faucet has the whitelist as an optional argument, not a mandatory one. I suggest to merge this and fix it by merging master into community-sidechain branch

**Comment by brenzi** [comment]
above commit should actually fix it and minimize the diff to master. merged it with master:
https://github.com/encointer/encointer-node/compare/6f5d7bf1275823bcbf2e5d9a17a8fdd2fb9efe11..szp/community-sidechain-update_2

**Pull request merged**